### ISSUE TITLE
Cross-Validation is a Bad idea in Deep Learning?

### ISSUE THREAD

**Issue by acpn** [BVLC/caffe#3405]
Hi guys,

I'm trying do cross-validation in caffe because I've a small set of images 3K, I'm do it this way:

split my train images in 10 subsets, then I generate the lmdb files, after this I train in subsets 1 to 9 and validation in 10, after this I finetuning this trained network in subsets 2 to 10 an validate in subset 1, and subsequently until pass for all subsets.

The problem is which the network suffers of overfiting because when I do finetuning my subset of validation already exposed to training phase.

So what I'm doing wrong? Is this way which do cross-validation in caffe?

I read about this in some places like github and here in caffe user too, but isn't clear.

If anyone help me I really thanks, I've lost a week on this and the problem persist.

Thanks guys.

**Comment by seanbell** [comment]
This is a general machine-learning question, not a caffe question.  If I'm understanding your description correctly, you're not doing cross-validation correctly, and you've contaminated training and validation sets.  That won't cause overfitting (and it won't prevent it either) -- it just means your validation error is meaningless.

You probably just want to freeze the lower layers (set learning rate to 0) and only learn the last 1-3 layers (for example).  That's a better solution to overfitting than your scheme (which doesn't make sense).

Also, from https://github.com/BVLC/caffe/blob/master/CONTRIBUTING.md:
> _Please do not post usage, installation, or modeling questions, or other requests for help to Issues._
Use the [caffe-users list](https://groups.google.com/forum/#!forum/caffe-users) instead. This helps developers maintain a clear, uncluttered, and efficient view of the state of Caffe.
